### PR TITLE
Global data new

### PR DESCRIPTION
**Pull request by Christian-B**
Part of https://github.com/SpiNNakerManchester/SpiNNUtils/pull/172

### CODE REVIEW

**Comment by Christian-B** [comment]
Fixes https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/issues/526

File: spinn_front_end_common/interface/java_caller.py [hunk 1]
```diff
@@ -219,12 +184,12 @@ def set_report_folder(self, report_folder):
...
         self._report_folder = report_folder
 
-    def set_placements(self, placements, transceiver):
+    def set_placements(self, used_placements):
         """ Passes in the placements leaving this class to decide pass it to
             Java.
 
         This method may obtain extra information about he placements which is
-        why it also needs the transceiver.
+        why it also accesses the transceiver.
```
Comment (andrewgait) [inline spinn_front_end_common/interface/java_caller.py:192]: I guess it does access the transceiver through later functions, but this function itself doesn't actually access it, so perhaps this whole comment is unnecessary.

File: spinn_front_end_common/utilities/report_functions/router_collision_potential_report.py [hunk 2]
```diff
@@ -16,23 +16,17 @@
 import os
 from collections import defaultdict
 from spinn_machine.router import Router
-from spinn_front_end_common.utilities.globals_variables import (
-    report_default_directory)
+from spinn_front_end_common.data import FecDataView
 
 COLLISION_REPORT = "routing_collision_protential_report.rpt"
```
Comment (andrewgait) [inline spinn_front_end_common/utilities/report_functions/router_collision_potential_report.py:21]: while I'm in here, can we rename this report to change "protential" to "potential"?

**Review by andrewgait — CHANGES_REQUESTED** [review]
Looks fine in general - again just some minor edits and a few questions.

File: spinn_front_end_common/data/__init__.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2021 The University of Manchester
+# Copyright (c) 2021-2022 The University of Manchester
```
[the remark this answers is not in the document]
Comment (Christian-B) [reply spinn_front_end_common/data/__init__.py:1]: Agreed. No idea why git did that  But little can be done.

File: spinn_front_end_common/data/fec_data_view.py [hunk 4]
```diff
@@ -0,0 +1,1065 @@
...
+            # Throws OSError or FileExistsError (a subclass of OSError) if the
+            # directory exists.
+            os.makedirs(child)
+        elif not os.path.exists(child):
+            cls._make_dirs(child)
+        return child
+
+    @staticmethod
+    def _make_dirs(path):
+        # Workaround for Python 2/3 Compatibility (Python 3 raises on exists)
```
[the remark this answers is not in the document]
Comment (Christian-B) [reply spinn_front_end_common/data/fec_data_view.py:748]: nope removed

File: spinn_front_end_common/interface/abstract_spinnaker_base.py [hunk 5]
```diff
@@ -973,40 +580,38 @@ def _run(self, run_time, sync_time):
...
-        self._status = Simulator_Status.FINISHED
 
     def _is_per_timestep_sdram(self):
-        for placement in self._placements.placements:
+        for placement in self._data_writer.iterate_placemements():
             if placement.vertex.resources_required.sdram.per_timestep:
                 return True
         return False
 
     def _add_commands_to_command_sender(self):
```
[the remark this answers is not in the document]
Comment (Christian-B) [reply spinn_front_end_common/interface/abstract_spinnaker_base.py:611]: https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/issues/957

File: spinn_front_end_common/interface/config_handler.py [hunk 6]
```diff
@@ -368,85 +225,6 @@ def write_errored_file(self):
...
-        :rtype: int
-        """
-        return self.__time_scale_factor
-
-    @time_scale_factor.setter
-    def time_scale_factor(self, new_value):
-        self.__time_scale_factor = new_value
-
     @staticmethod
     def _make_dirs(path):
```
[the remark this answers is not in the document]
Comment (Christian-B) [reply spinn_front_end_common/interface/config_handler.py:229]: You thought right. Removed.

File: spinn_front_end_common/interface/interface_functions/host_no_bitfield_router_compression.py [hunk 7]
```diff
@@ -237,31 +147,32 @@ def _load_routing_table(self, table):
...
         """
+        transceiver = FecDataView.get_transceiver()
         for core_subset in executable_targets.all_core_subsets:
             x = core_subset.x
             y = core_subset.y
             for p in core_subset.processor_ids:
                 # Read the result from specified register
                 if self.__result_register == 0:
-                    result = transceiver.read_user_0(x, y, p)
+                    result = transceiver().read_user_0(x, y, p)
```
[the remark this answers is not in the document]
Comment (Christian-B) [reply spinn_front_end_common/interface/interface_functions/host_no_bitfield_router_compression.py:175]: incorrect brackets removed.

File: spinn_front_end_common/interface/java_caller.py (hunk 1 above)
Comment (Christian-B) [reply spinn_front_end_common/interface/java_caller.py:192]: confusing comment removed.

File: spinn_front_end_common/utilities/report_functions/router_collision_potential_report.py (hunk 2 above)
Comment (Christian-B) [reply spinn_front_end_common/utilities/report_functions/router_collision_potential_report.py:21]: https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/issues/891

File: unittests/data/manual_check.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from spinn_utilities.exceptions import NotSetupException, DataNotYetAvialable
+from spinn_front_end_common.data import FecDataView
+from spinn_front_end_common.data.fec_data_writer import FecDataWriter
+from spinn_front_end_common.interface.config_setup import add_spinnaker_cfg
+from spinn_utilities.config_holder import clear_cfg_files
+
+# This can not be a unittest as the unitest suite would use the same
+# python console and therefor the same singleton multiple times
```
[the remark this answers is not in the document]
Comment (Christian-B) [reply unittests/data/manual_check.py:23]: All therefor changed to therefore

**Review by andrewgait — APPROVED** [review]

**Pull request merged**